CONTRIBUTING.rst

```diff
@@ -20,6 +20,8 @@ The following pull request labels are recognized:
 * dependencies: Update one or more dependencies version
 * performance: Improve performance of an existing feature
 
+Note: Changelog entries are created for each commit, so please rebase/squash before merge.
```

Comment:
hm, auto should do them per PR not per commit

Reply:
It didnt commit the changelog, but you can see it at least part of it in the release action https://github.com/con/duct/actions/runs/10497786670/job/29081383352

I've been committing sloppy, so we slammed it :/

```
✔  success   Calculated version bump: minor
ℹ  info      New Release Notes
 #### 🚀 Enhancement

- [DATALAD RUNCMD] Autoupdate readme [#159](https://github.com/con/duct/pull/159) ([@asmacdo](https://github.com/asmacdo))
- Update src/con_duct/__main__.py [#159](https://github.com/con/duct/pull/159) ([@asmacdo](https://github.com/asmacdo))
- Order from nothing to most [#159](https://github.com/con/duct/pull/159) ([@asmacdo](https://github.com/asmacdo))
- Add log level NONE and deprecate quiet [#159](https://github.com/con/duct/pull/159) ([@asmacdo](https://github.com/asmacdo))
- Test that values are truthy, not None is insufficient [#153](https://github.com/con/duct/pull/153) ([@asmacdo](https://github.com/asmacdo))
- Pass USER env var to tox env so we can test it is set [#153](https://github.com/con/duct/pull/153) ([@asmacdo](https://github.com/asmacdo))
- Test that values are set in system_info [#153](https://github.com/con/duct/pull/153) ([@asmacdo](https://github.com/asmacdo))
- Collect hostname in sys info [#153](https://github.com/con/duct/pull/153) ([@asmacdo](https://github.com/asmacdo))
- Fixup: typing tests [#141](https://github.com/con/duct/pull/141) ([@asmacdo](https://github.com/asmacdo))
- Fixup: rm whitespace [#141](https://github.com/con/duct/pull/141) ([@asmacdo](https://github.com/asmacdo))
- Update src/con_duct/__main__.py [#141](https://github.com/con/duct/pull/141) ([@asmacdo](https://github.com/asmacdo))
- Replace cached_property with regular property [#141](https://github.com/con/duct/pull/141) ([@yarikoptic](https://github.com/yarikoptic))
- BF(TST)+RF: make execution_summary robust to case where Report is not yet complete [#141](https://github.com/con/duct/pull/141) ([@yarikoptic](https://github.com/yarikoptic))
- RF+BF: make explicit wall_clock_time separate from elapsed_time [#141](https://github.com/con/duct/pull/141) ([@yarikoptic](https://github.com/yarikoptic))
- fixup: helptext [#140](https://github.com/con/duct/pull/140) ([@asmacdo](https://github.com/asmacdo))
- [DATALAD RUNCMD] Autoupdate readme [#140](https://github.com/con/duct/pull/140) ([@asmacdo](https://github.com/asmacdo))
- Update src/con_duct/__main__.py [#140](https://github.com/con/duct/pull/140) ([@asmacdo](https://github.com/asmacdo))
- Add --quiet back in [#140](https://github.com/con/duct/pull/140) ([@asmacdo](https://github.com/asmacdo))
- Fixup: Set the log level [#140](https://github.com/con/duct/pull/140) ([@asmacdo](https://github.com/asmacdo))
- Default info level [#140](https://github.com/con/duct/pull/140) ([@yarikoptic](https://github.com/yarikoptic))
- BF(TST): fixup the test for changes [#140](https://github.com/con/duct/pull/140) ([@yarikoptic](https://github.com/yarikoptic))
- BF: use caplog to verify presence of desired report in the last log message [#140](https://github.com/con/duct/pull/140) ([@yarikoptic](https://github.com/yarikoptic))
- RF: Add logging, dissolve duct_print (INFO level), add CLI option -l, dissolve --quiet [#140](https://github.com/con/duct/pull/140) ([@yarikoptic](https://github.com/yarikoptic))
- ENH: give "typical" shell behavior when command is not found to be executed [#138](https://github.com/con/duct/pull/138) ([@yarikoptic](https://github.com/yarikoptic))
- Use None rather than 0 prior to measurement [#135](https://github.com/con/duct/pull/135) ([@asmacdo](https://github.com/asmacdo))
- RF+ENH: output messages to stderr not stdout + move printing out of "controllers/models" [#136](https://github.com/con/duct/pull/136) ([@yarikoptic](https://github.com/yarikoptic))
- Remove unused/untested print_max_values [#136](https://github.com/con/duct/pull/136) ([@yarikoptic](https://github.com/yarikoptic))
- consolidate and parameterize tests [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- Explicitly show that ps gives memory in KiB [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- Consolodate number checking [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- [DATALAD RUNCMD] Autoupdate readme [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- shorten line, ignored by black, but flake8 failed [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- rerun pre-commit -a [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- fixup mypy [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- fix map usage [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- fixup rm execution formatting, must be ok with 'unknown' [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- type check func [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- Add rounding and units for user-facing (rendered) summary [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- execution summary file should contain raw values [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- Add ts type validation [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- hit all the fields for coverage [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- Ensure averages stay numbers [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- fixup sample total field, no longer kib [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- Add ProcessStats validation [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- fixup comment [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- Switch to bytes [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- [DATALAD RUNCMD] Autoupdate reademe [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- Remove units for machine readability [#125](https://github.com/con/duct/pull/125) ([@asmacdo](https://github.com/asmacdo))
- Sweep through possible exit codes [#119](https://github.com/con/duct/pull/119) ([@yarikoptic](https://github.com/yarikoptic))
- Make execute return returncode of the process and use it for duct CLI process exit code [#119](https://github.com/con/duct/pull/119) ([@yarikoptic](https://github.com/yarikoptic))

#### 🐛 Bug Fix

- Add direct pytest usage to CONTRIBUTING [#161](https://github.com/con/duct/pull/161) ([@asmacdo](https://github.com/asmacdo))
- [DATALAD RUNCMD] Autoupdate readme [#158](https://github.com/con/duct/pull/158) ([@asmacdo](https://github.com/asmacdo))
- Update __main__.py [#158](https://github.com/con/duct/pull/158) ([@asmacdo](https://github.com/asmacdo))
- fixup wrapping [#158](https://github.com/con/duct/pull/158) ([@asmacdo](https://github.com/asmacdo))
- Apply code review suggestions to source material [#158](https://github.com/con/duct/pull/158) ([@asmacdo](https://github.com/asmacdo))
- Apply suggestions from code review [#158](https://github.com/con/duct/pull/158) ([@asmacdo](https://github.com/asmacdo))
- fixup mypy [#158](https://github.com/con/duct/pull/158) ([@asmacdo](https://github.com/asmacdo))
- fixup line wrap [#158](https://github.com/con/duct/pull/158) ([@asmacdo](https://github.com/asmacdo))
- Extend helptext introduction [#158](https://github.com/con/duct/pull/158) ([@asmacdo](https://github.com/asmacdo))
- Improve helptext top-level description [#158](https://github.com/con/duct/pull/158) ([@asmacdo](https://github.com/asmacdo))
- Check that each PR has one of the semver labels [#156](https://github.com/con/duct/pull/156) ([@asmacdo](https://github.com/asmacdo))
- Relax time requirement to prevent flake [#155](https://github.com/con/duct/pull/155) ([@asmacdo](https://github.com/asmacdo))
- Do not use setsid directly, use dedicated start_new_session [#155](https://github.com/con/duct/pull/155) ([@yarikoptic](https://github.com/yarikoptic))
- Disable macos-latest also [#151](https://github.com/con/duct/pull/151) ([@asmacdo](https://github.com/asmacdo))
- Disable MacOS tests [#151](https://github.com/con/duct/pull/151) ([@asmacdo](https://github.com/asmacdo))
- Sanity assertions for execution summary [#151](https://github.com/con/duct/pull/151) ([@asmacdo](https://github.com/asmacdo))
- [DATALAD RUNCMD] Autoupdate readme [#151](https://github.com/con/duct/pull/151) ([@asmacdo](https://github.com/asmacdo))
- Fix typo in pmem calculation [#151](https://github.com/con/duct/pull/151) ([@asmacdo](https://github.com/asmacdo))
- Relax execution requirement to 0.2 [#152](https://github.com/con/duct/pull/152) ([@asmacdo](https://github.com/asmacdo))
- Collect sys info and env in parallel [#152](https://github.com/con/duct/pull/152) ([@asmacdo](https://github.com/asmacdo))
- Revert back to original smon + blacken and logging [#147](https://github.com/con/duct/pull/147) ([@asmacdo](https://github.com/asmacdo) [@yarikoptic](https://github.com/yarikoptic))
- [DATALAD RUNCMD] Autoupdate readme [#146](https://github.com/con/duct/pull/146) ([@asmacdo](https://github.com/asmacdo))
- cleanup comments [#146](https://github.com/con/duct/pull/146) ([@asmacdo](https://github.com/asmacdo))
- RF+BF: update maxes on each sample, more logging during monitoring [#146](https://github.com/con/duct/pull/146) ([@yarikoptic](https://github.com/yarikoptic))
- Use Disable instead of Suppress for quiet [#146](https://github.com/con/duct/pull/146) ([@yarikoptic](https://github.com/yarikoptic))
- RF(TESTS): to avoid duplicated specification of Arguments [#146](https://github.com/con/duct/pull/146) ([@yarikoptic](https://github.com/yarikoptic))
- Centralize assert for expected files [#146](https://github.com/con/duct/pull/146) ([@yarikoptic](https://github.com/yarikoptic))
- RF: no shebang since file is no longer can be executed [#139](https://github.com/con/duct/pull/139) ([@yarikoptic](https://github.com/yarikoptic))

#### Authors: 2

- Austin Macdonald ([@asmacdo](https://github.com/asmacdo))
- Yaroslav Halchenko ([@yarikoptic](https://github.com/yarikoptic))
✖  error     Received E2BIG from execSync.

This usually occurs when the argument list is too large for the command you are trying to run. 

Please consider disabling your 'auto-exec' usage and following this issue for updates: https://github.com/intuit/auto/issues/1294
```

Reply:
something is odd -- again, `auto` should have picked up PR titles not individual commits.  Were these commits pushed directly without PRs or part of the PRs?

Reply:
I haven't pushed outside of a PR since the initial hackout. Thats not it (also FWIW i copied the .autorc from datalad-installer, and indeed, the occasional PR with multiple commits only gets 1 changelog entry.

Reply:
@yarikoptic started a discussion on the auto repo 

https://github.com/intuit/auto/discussions/2473

Reply:
@jwodder might know

Reply:
It's because you're rebase-merging the PRs instead of merging them normally.

Reply:
That would make sense, I'll switch back for the next release and prove it.